vortex-ffi/examples/hello_vortex.rs

```diff
@@ -0,0 +1,108 @@
+#![allow(clippy::unwrap_used, clippy::expect_used)]
+//! This example shows usage of the Vortex C FFI to read a Vortex file written by a Rust client.
+//!
+//! You can invoke this example from a checkout by running
+//!
+//! ```ignore
+//!cargo run -p vortex-ffi --example hello_vortex
+//! ```
+
+use std::env;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use std::sync::LazyLock;
+
+use tempfile::NamedTempFile;
+use tokio::fs::File as TokioFile;
+use tokio::runtime::Runtime;
+use vortex::arrays::{ChunkedArray, StructArray};
+use vortex::buffer::Buffer;
+use vortex::error::VortexResult;
+use vortex::file::VortexWriteOptions;
+use vortex::{Array, ArrayRef, IntoArray};
+
+static RUNTIME: LazyLock<Runtime> = LazyLock::new(|| Runtime::new().unwrap());
+
+const BIN_NAME: &str = "hello_vortex";
+
+pub fn main() -> VortexResult<()> {
+    let bin_path = PathBuf::new()
+        .join(env!("CARGO_MANIFEST_DIR"))
+        .join("examples")
+        .join(BIN_NAME);
+
+    let exit = Command::new("cc")
```

Comment:
i originally tried using the `cc` crate but that doesn't let you build static executables, only .so's